### ISSUE TITLE
False Positive for "Unnecessary semicolon" rule

### ISSUE THREAD

**Issue by Bastian** [pinterest/ktlint#957]
Having an empty enum class like 
```kotlin
enum class Test {
    ;
    abstract val field: String
}
```
causes a false-positive in ktlint:
![ktlint-bug](https://user-images.githubusercontent.com/5033001/97866166-b0069800-1d0b-11eb-9295-cf7f4c865076.png)

## Your Environment
* Version of ktlint used: 0.39.0
* Kotlin version 1.4.10-release-411 (JRE 1.8.0_265-b01)
